src/analysis/souffle/tags.dl

```diff
@@ -28,4 +28,11 @@
 .type Tag <: symbol
 .decl isTag(tag: Tag)
 
+// An access path marked with an integrity tag (shortened here to IntegTag)
+// indicates that some property must be true of the data at the given
+// AccessPath. The conservative over-approximation for propagating an integrity
+// tag is that a node has an integrity tag only if all of its predecessors have
+// that integrity tag.
+.type IntegrityTag <: symbol
```

Comment:
I am not sure if there is extra benefit with keeping a separate type. We should simply use the `Tag` type?

Reply:
The reason why I think it makes sense to have two types is because it is entirely possible that someone, while hacking on the Datalog or code that generates Datalog, could accidentally write a rule body that propagates an integrity tag based upon a fact involving a confidentiality tag or vice-versa. This separate type helps prevent that sort of error from occuring.

Reply:
Isn't the propagation of tags internal to the Raksha implementation? I don't see how that mistake would happen. The programmer writes rules that add or remove tags through a predicate that mentions whether the tag is added as confidentiality or integrity. So which predicate is used makes it clear.

Reply:
I see your point about unintentional errors,. However, I think we should be able to treat a tag as either a confidentiality or an integrity tag. Introducing a new type keeps the namespaces of confidentiality and integrity separate, which I think is not quite desirable.

That said, I guess we could start with a separate type and change it later if needed. It might be harder (but not impossible) to go the other way. So, OK to leave it for now while we understand the problem. However, I think we will eventually want to unify them.

Reply:
Sounds good, will leave it for now, and will remove distinction if it causes unwanted friction.